完善安装功能

YvonneOfficial opened the pull request:
我完善了v4的整个安装功能
程序会在YvonneOfficial/Temp-Resources仓库内获取服务器列表和服务器配置
自动检测是否有重复的配置项 重复则不安装
下载时自动选择最优线路 自动选择Github和镜像源
安装完毕后自动添加只读 防止部分模组自动添加服务器

## Sourcery 总结

通过从多个来源并发加载服务器列表和配置、自动选择最佳下载路径以及管理重复项，增强了私服安装功能。集成了带有加载指示器和动态复选框的响应式用户界面，通过只读管理安全地更新 regionInfo.json，并连接了导航和进度消息，以提供无缝的安装体验。

新功能：
- 引入 `ServerLoader` 和 `ServerConfigLoader`，用于从 GitHub 和镜像源并行获取服务器列表和配置
- 添加动态的 `PrivateServerInterface UI`，用于在数据获取完成后显示加载状态并填充服务器复选框
- 实现完整的安装工作流，该工作流加载服务器配置、检测并跳过重复项，并通过适当的只读处理更新 `regionInfo.json`

增强功能：
- 在获取数据时自动选择更快的下载源，并在需要时回退到镜像
- 重构导航，使其包含可从主窗口和主页界面访问的私服安装页面
- 改进日志记录和消息回调，以在安装过程中报告进度、成功、警告和错误

<details>
<summary>Original summary in English</summary>

## Sourcery 总结

实现具有动态用户界面和健壮文件处理的并发私服安装工作流

新功能：
- 添加 ServerLoader 和 ServerConfigLoader，以并发方式从 GitHub 和镜像源获取服务器列表和配置
- 引入响应式 PrivateServerInterface，具有加载指示器、动态复选框和基于信号的状态消息
- 实现完整的安装过程，该过程获取配置、跳过重复项、更新 regionInfo.json 并管理文件只读属性

改进：
- 自动选择最快的下载源，并回退到镜像
- 将新的“私服安装”页面集成到导航中，并通过信号启用 homeInterface 驱动的导航
- 使用 InfoBar 消息和优化的回调，集中化日志记录和进度反馈

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Implement concurrent private server installation workflow with dynamic UI and robust file handling

New Features:
- Add ServerLoader and ServerConfigLoader to fetch server lists and configurations concurrently from GitHub and mirror sources
- Introduce a responsive PrivateServerInterface with loading indicators, dynamic checkboxes, and signal-based status messages
- Implement full install process that fetches configurations, skips duplicates, updates regionInfo.json, and manages file readonly attributes

Enhancements:
- Automatically select the fastest download source with fallback to a mirror
- Integrate a new '私服安装' page into navigation and enable homeInterface-driven navigation via signals
- Centralize logging and progress feedback using InfoBar messages and refined callbacks

</details>

</details>

YvonneOfficial committed:
添加自动获取服务器列表和主页卡片导航
添加了自动检测服务器列表（虽然这个库是临时的
主页的卡片添加了导航，点击后自动跳转到指定页
爱来自忆梦

YvonneOfficial committed:
feat(server): 实现多源服务器配置加载与安装功能

- 添加从GitHub和镜像源双重加载机制，提高服务器配置加载可靠性
- 实现ServerConfigLoader类，支持并发加载多个服务器配置
- 优化安装流程，使用线程处理服务器配置获取和文件写入操作
- 添加regionInfo.json文件的只读属性处理，确保安装后文件权限正确
- 实现重复服务器配置检测，避免重复安装相同配置
- 改进错误处理机制，提供更详细的错误信息和日志记录
- 更新界面交互逻辑，确保安装过程中按钮状态正确更新
- 修复默认服务器配置启用状态问题
- 优化信息提示显示，区分成功、警告和错误状态的消息展示

YvonneOfficial committed:
更新链接

YvonneOfficial committed:
Merge branch 'v4' into v4-dev

YvonneOfficial committed:
refactor(utils): 重构工具函数模块

- 移除未使用的导入模块
- 将服务器相关功能移至独立的serverManager模块
- 优化服务器状态切换逻辑
- 改进服务器配置加载方式
- 更新模块导入路径和依赖关系
- 优化列表推导式为显式循环以提高可读性

QingFeng-awa reviewed (changes requested):
`logMessage`已被弃用，将很快被移除，且实际效果与`logInfo`等价。
请使用`logInfo`或`logDebug`代替。

YvonneOfficial committed:
refactor(log): 统一使用logInfo替换logMessage

- 将serverManager.py中的logMessage调用替换为logInfo
- 更新privateServerInterface.py中的日志记录方法
- 修改导入语句以使用logInfo函数
- 确保所有信息级别日志使用统一接口

QingFeng-awa reviewed (approved):
不认为一键替换为`logInfo`是个聪明的选择。
我会新打开一个 PR 来解决此问题。

Hiraeth-Wave commented on app/view/privateServerInterface.py:
```diff
@@ -84,34 +91,143 @@ def initWidget(self):
+        
+    def onLoadFailed(self, error_msg):
+        """服务器列表加载失败"""
+        logInfo(f"服务器列表加载失败: {error_msg}")
```
你当真觉得这里该用 Info？
